scheduler/core/sources.py

```diff
@@ -101,15 +101,15 @@ def use_file(self,
                 if calendar and gmos_fpu and gmos_gratings:
                     file_resource = FileResourceService()
 
-                    # TODO: files_input is undefined. This will be fixed later.
-                    for site in files_input.sites:
+                    for site in sites:
                         suffix = ('s' if site == Site.GS else 'n').upper()
-                        file_resource.load_files(f'GMOS{suffix}_fpu_barcode.txt',
+                        file_resource.load_files(site,
+                                                 f'GMOS{suffix}_fpu_barcode.txt',
                                                  gmos_fpu,
                                                  gmos_gratings,
                                                  calendar)
```

Comment:
In the most updated code, `calendar` no longer gets passed to `load_files` because it was causing the calendar to load twice (once for each site), although now I think it might make sense to "load" it twice and during each load, process the sheet for the current site... so let me look at this, fix the test cases, and get back to you.